code/game/objects/items/cards_ids.dm

```diff
@@ -757,3 +757,30 @@ update_label("John Doe", "Clowny")
 
 /obj/item/card/id/job/lawyer
 	icon_state = "lawyer"
+
+/obj/item/card/id/pass
+	name = "promotion pass"
+	desc = "A card that, when swiped on your ID card, will grant you all the access. Should not substitute your actual ID card."
+	icon_state = "data_1"
+	registered_name = "Unregistered ID"
+	assignment = "Access Pass"
+
+/obj/item/card/id/pass/afterattack(atom/target, mob/user, proximity)
+	. = ..()
+	if(istype(target,/obj/item/card/id) && proximity)
+		var/obj/item/card/id/idcard = target
+		for(var/ac in access)
+			idcard.access |= ac
+		if(assignment!=initial(assignment))
+			idcard.assignment = assignment
+		if(name!=initial(name))
+			idcard.name = name
+		to_chat(user, "You upgrade your [idcard] with the [name].")
+		log_id("[key_name(user)] added access to '[idcard]' using [src] at [AREACOORD(user)].")
+		qdel(src)
+
+/obj/item/card/id/pass/minisec
```

Comment:
Doesn't remove the original item from deputy.dm?

Reply:
https://github.com/BeeStation/BeeStation-Hornet/blob/94e9c1a1b70aa9753388b57df8c232ecdb117046/code/modules/jobs/job_types/deputy.dm#L45